pkg/controller/kogitoapp/resource/build_config.go

```diff
@@ -39,44 +39,50 @@ const (
 	tagLatest          = "latest"
 	// ImageStreamNamespace default namespace for the ImageStreams
 	ImageStreamNamespace = "openshift"
+	// ImageStreamLocation if no Kogito images found, use this to fetch the imagestream and import it on cluster under the default namespace
+	ImageStreamLocation = "https://raw.githubusercontent.com/kiegroup/kogito-cloud/%s/s2i/kogito-imagestream.yaml"
```

Comment:
I am a bit worried of maintainability of these hardcoded values, specially if we will productize it.

Reply:
I'm also worried. Do you have a suggestion for this particular case? I mean, those should follow the tag/release from the `kogito-cloud` repository to not break old operators installations.

Reply:
I think that we may use config files to define image names and tags, this way we can easily supply different values if needed. Also it will remove external dependency on GitHub (useful for cases when cluster is in offline environment).

Reply:
The removal of the Github reference is a must. I'd say that the Image Stream should be created like we do today with other resources, but not in the "resource creation fase", before.

[Also we already have an input for the image tags](https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/apis/app/v1alpha1/kogitoapp_types.go#L88-L90) that should be used as a reference to create the IS. If it's not given, we just use the default version which is the one from the operator itself.

Reply:
@ricardozanini @mswiderski @sutaakar ^

Reply:
I’d say that given version of operator should support only given version of the images. Otherwise we could run out of sync

Another question - in what namespace will these image streams be installed? Openshift or default or?

Reply:
@mswiderski the operator will first look for the imagestreams on the openshift namespace and in the namespace you are creating the kogito app.

If no imagestream are found, the operator will create it on the current namespace.

Reply:
not really, IIRC the first place it will be looking is on the local namespace then openshift.
Only if all imagestreams are available on openshift namespace we set the Instance.Namespace to openshift.

@ricardozanini can confirm it.

Reply:
@mswiderski yeah, right, that is the current behaviour. with this PR it will change: https://github.com/kiegroup/kogito-cloud-operator/pull/70/files#diff-4be0017824908f3c2c4610bbd4bfef96L55

by default it will first look on the current namespace.
Only if all image stream are available on openshift namespace, we set it manually.

Reply:
@mswiderski the default behavior is to look to openshift namespace, unless you specify the image details in the CLI via: `--image-s2i=mynamespace/kogito-quarkus-ubi8-s2i:latest`

You can run `kogito deploy --help` to see more details about this option.

Also, you can easily set this via CR: `.Spec.Build.ImageS2i`.

@spolti I guess you're considering those options too?

Reply:
@spolti looking at the code you linked above, why are you removing the ns?

Reply:
@ricardozanini I think we spoke about it a few days ago, I removed it as part of this change.
only if all imagestreams are available on openshift namespace I set it manually here: https://github.com/kiegroup/kogito-cloud-operator/pull/70/files#diff-06053a5d4cb74bb9cf01187cb724f70dR423

this way, if the IS are created, it will look on the current namespace.

Reply:
@spolti Ok for me to use GitHub IS as a temporary solution.

Reply:
> @mswiderski the default behavior is to look to openshift namespace, unless you specify the image details in the CLI via: `--image-s2i=mynamespace/kogito-quarkus-ubi8-s2i:latest`

in that case we should always install image streams in openshift namespace as it is used by default and most users won't bother to use the `--image-s2i` argument

Reply:
Github reference removed.